[lldb][swift] Make swiftTest decorator not always skip and skipIf regressed tests temporarily

Teemperor opened the pull request:
_get_bool_config_skip_if_decorator returns a decorator, but the
is_not_swift_compatible function is supposed to return a bool. The skipTestIfFn
is supposed to create the decorator from the function we define.

As _get_bool_config_skip_if_decorator's return value is just interpreted as True,
this ends up causing that we skip all Swift tests all the time.

This just adds a _get_bool_config() function that actually returns a bool.
We could probably also solve this by doing a function composition of the
returned decorator and the decorator we create from our local function, but
just having a bool seems less likely to break.

Also re-enables and skipIf's the tests that are crashing/failing since the commit landed for now.

Teemperor committed:
[lldb][swift] Make swiftTest decorator not always skip

_get_bool_config_skip_if_decorator returns a decorator, but the
is_not_swift_compatible function is supposed to return a bool. The skipTestIfFn
is supposed to create the decorator from the function we define.

As _get_bool_config_skip_if_decorator's return value is just interpreted as True,
this ends up causing that we skip all Swift tests all the time.

This just adds a _get_bool_config() function that actually returns a bool.
We could probably also solve this by doing a function composition of the
returned decorator and the decorator we create from our local function, but
just having a bool seems less likely to break.

Teemperor commented:
@swift-ci test

JDevlieghere commented on lldb/packages/Python/lldbsuite/test/decorators.py:
```diff
@@ -886,11 +885,14 @@ def skipIfAsan(func):
     """Skip this test if the environment is set up to run LLDB *itself* under ASAN."""
     return skipTestIfFn(is_running_under_asan)(func)
 
-def _get_bool_config_skip_if_decorator(key):
```
Should this part be upstream?

Teemperor replied:
Sounds good to me. I think we anyway need to wait with this until we tracked down the regressions, I'll upstream that and then rebase this on top.

JDevlieghere replied:
Thanks, it's more about making sure it doesn't get forgotten than blocking the merge.

kastiglione commented on lldb/packages/Python/lldbsuite/test/decorators.py:
```diff
@@ -886,11 +885,14 @@ def skipIfAsan(func):
-    have = value_node.GetValueForKey("value").GetBooleanValue(fail_value)
+    return value_node.GetValueForKey("value").GetBooleanValue(fail_value)
+
+def _get_bool_connfig_skip_if_decorator(key):
```
```suggestion
def _get_bool_config_skip_if_decorator(key):
```

Teemperor replied:
The question is now: How did this still do the right thing?

kastiglione replied:
it didn't work for me, I only noticed this because of python exceptions

Teemperor replied:
Interesting, because the tests on swift CI actually did run as expected: https://ci.swift.org/job/apple-llvm-project-pr-linux/1530/console

Teemperor commented:
@swift-ci test

Teemperor committed:
[lldb][swift] Skip all failing tests

Adds skipIf to all the tests that regressed since 2c911bceb06ed376801251bdfd992905a66f276c
landed and disabled the Swift tests unintentionally.

Teemperor commented:
@swift-ci test

Teemperor commented:
@swift-ci test

Teemperor commented:
@swift-ci test

adrian-prantl commented:
@Teemperor please don't forget to cherry-pick this to swift/next

adrian-prantl commented:
> @Teemperor please don't forget to cherry-pick this to swift/next

but don't cherry-pick the XFAILs
